docs/rules/number-literal-case.md

```diff
@@ -17,50 +17,148 @@ Differentiating the casing of the identifier and value clearly separates them an
 [Hexadecimal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Hexadecimal)
 
 ```js
+// ❌
 const foo = 0XFF;
 const foo = 0xff;
 const foo = 0Xff;
 const foo = 0Xffn;
+
+// ✅
+const foo = 0xFF;
+const foo = 0xFFn;
 ```
 
 [Binary](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Binary)
 
 ```js
+// ❌
 const foo = 0B10;
 const foo = 0B10n;
+
+// ✅
+const foo = 0b10;
+const foo = 0b10n;
 ```
 
 [Octal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Octal)
 
 ```js
+// ❌
 const foo = 0O76;
 const foo = 0O76n;
+
+// ✅
+const foo = 0o76;
+const foo = 0o76n;
 ```
 
-Exponential notation
+[Exponential notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Exponential)
 
 ```js
+// ❌
 const foo = 2E-5;
+const foo = 2E+5;
+const foo = 2E5;
+
+// ✅
+const foo = 2e-5;
+const foo = 2e+5;
+const foo = 2e5;
 ```
 
-## Pass
+## Options
 
-```js
-const foo = 0xFF;
-```
+Type: `object`
```

Comment:
I think we can also allow `'lowercase'`/`'uppercase'` here. `'lowercase'` means everything should be lowercased.

Reply:
Make sense.